lib/battle/CBattleInfoCallback.cpp

```diff
@@ -855,6 +855,24 @@ AccessibilityInfo CBattleInfoCallback::getAccesibility() const
 		ret[BattleHex(0, y)] = EAccessibility::SIDE_COLUMN;
 	}
 
+	//special battlefields with logically unavailable tiles
+	std::vector<BattleHex> impassableHexes;
+	if(battleGetBattlefieldType().num == BFieldType::SHIP_TO_SHIP)
+		impassableHexes = 
+		{ 

```

Comment:
What about:
`for(auto group : groups)
	for(auto item : group)
		doSth();`
Or:
if(a)
	if(b)
		doSth();

I got used to skip { } in cases of multi conditional expressions, with exception when there is "else" so it will be "sealed" by curly brackets to which "if" it belongs.